language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/LSPackageLoader.java

```diff
@@ -70,69 +97,156 @@ public static LSPackageLoader getInstance(LanguageServerContext context) {
     }
 
     private LSPackageLoader(LanguageServerContext context) {
+//        this.userHome = userHome;
         this.clientLogger = LSClientLogger.getInstance(context);
+        this.centralPackageDescriptorLoader = CentralPackageDescriptorLoader.getInstance(context);
         context.put(LS_PACKAGE_LOADER_KEY, this);
     }
 
+    public boolean isInitialized() {
+        return Boolean.TRUE.equals(initialized);
+    }
+
     /**
-     * Get the local repo packages.
+     * Load modules from the Ballerina environment.
      *
-     * @return {@link List} of local repo packages
+     * @param context language server context.
      */
-    public List<ModuleInfo> getLocalRepoPackages(PackageRepository repository) {
-        if (!this.localRepoPackages.isEmpty()) {
-            return this.localRepoPackages;
-        }
-        this.localRepoPackages.addAll(checkAndResolvePackagesFromRepository(repository, Collections.emptyList(),
-                this.distRepoPackages.stream().map(ModuleInfo::packageIdentifier).collect(Collectors.toSet())));
-        return localRepoPackages;
+    public void loadModules(LanguageServerContext context) {
+            LSClientLogger lsClientLogger = LSClientLogger.getInstance(context);
+            String taskId = UUID.randomUUID().toString();
+            notificationTaskId = taskId;
+            Map<String, ModuleInfo> packagesList = new HashMap<>();
+            CompletableFuture.runAsync(() -> {
+                this.languageClient = context.get(ExtendedLanguageClient.class);
+                this.languageClient.showMessage(
+                        new MessageParams(MessageType.Info, "Indexing Ballerina packages: " +
+                                "some completions may not be available until indexing is complete."));
+                if (languageClient == null) {
+                    return;
+                }
```

Comment:
In this case, shouldn't the null check be before the invocation of the show message?

Reply:
Yes my mistake